core/lib/zksync_core/src/api_server/web3/metrics.rs

```diff
@@ -195,3 +195,61 @@ pub(super) struct PubSubMetrics {
 
 #[vise::register]
 pub(super) static PUB_SUB_METRICS: vise::Global<PubSubMetrics> = vise::Global::new();
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue, EncodeLabelSet)]
+#[metrics(label = "filter_type", rename_all = "snake_case")]
+pub(super) enum FilterType {
+    Events,
+    Blocks,
+    PendingTransactions,
+}
+
+impl From<TypedFilter> for FilterType {
```

Comment:
It make sense to implement `From<&TypedFilter>` instead; no data from the filter is required to construct `FilterType`.